### PR TITLE
Remove GoConvey

### PR DESCRIPTION
**Pull request by szh**
### What does this PR do?
- Rewrites test that used GoConvey to use a lightweight assertions library and native golang testing
- Refactor tests in `secretsyml/secretsyml_test.go` to reduce code duplication

### What ticket does this PR close?
Resolves N/A

#### Change log
- [ ] The CHANGELOG has been updated, or
- [x] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [x] This PR includes new unit and integration tests to go with the code changes, or
- [ ] The changes in this PR do not require tests

#### Documentation
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [x] This PR does not require updating any documentation

### CODE REVIEW

File: internal/command/action_test.go [hunk 1]
```diff
@@ -30,13 +30,13 @@ func TestConvertSubsToMap(t *testing.T) {
 
 		output := convertSubsToMap(input)
 
-		So(output, ShouldResemble, expected)
+		assert.EqualValues(t, expected, output)
```
Comment (diverdane) [inline internal/command/action_test.go:33]: Does `assert.EqualValues()` work for maps? (Reference: https://pkg.go.dev/github.com/stretchr/testify/assert#EqualValues)

_"EqualValues asserts that two objects are equal or convertable to the same types and equal."_

I take the meaning of `Values` here to mean the underlying value of an object after it has been converted to some type, and not the values in a map???

Wouldn't we need something like this?:
```
    assert.True(t, reflect.DeepEquals(expected, output))
```

**Review by diverdane — COMMENTED** [review]
Looks good! Just one question about how `assert.EqualValues()` works.

File: internal/command/action_test.go (hunk 1 above)
Comment (szh) [reply internal/command/action_test.go:33]: See my comment on the other PR here: https://github.com/cyberark/conjur-api-go/pull/118#discussion_r766900074

Comment (diverdane) [reply internal/command/action_test.go:33]: Wow, great to know!!!

**Review by diverdane — APPROVED** [review]
LGTM!!!

**Pull request merged**